src/makeInfinityAdder.js

```diff
@@ -5,6 +5,23 @@
  */
 function makeInfinityAdder() {
   // write code here
+  let sum = 0; // Initialize the sum to zero
+
+  function adder(...args) {
+    if (args.length === 0) {
+      const result = sum;
+
+      sum = 0; // Reset the sum after returning the result
+
+      return result;
+    }
+    // Sum up all the arguments passed
+    sum += args.reduce((acc, num) => acc + num, 0);
```

Comment:
The use of `reduce` here is unnecessary since the function is expected to receive only one argument at a time. You can simplify this by directly adding the single argument to the `sum` variable, after checking that the function was not called without arguments.